File: 20230426_Wednesday/20230426.rb
# Leetcode: 258. Add Digits.
# https://leetcode.com/problems/add-digits/
# = = = = = = = = = = = = = =
# Accepted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 91 ms, faster than 23.08% of Ruby online submissions for Add Digits.
# Memory Usage: 211.1 MB, less than 7.69% of Ruby online submissions for Add
# Digits.
# 2023.04.26 Daily Challenge.
# @param {Integer} num
# @return {Integer}
def add_digits(num)
  # Based on Leetcode's solution.
  return 0 if 0 == num
  return 9 if 0 == num % 9
  return num % 9
end

File: 20230426_Wednesday/20230426.go
// Leetcode: 258. Add Digits.
// https://leetcode.com/problems/add-digits/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 7 ms, faster than 12.58% of Go online submissions for Add Digits.
// Memory Usage: 2.1 MB, less than 98.11% of Go online submissions for Add
// Digits.
// 2023.04.26 Daily Challenge.

package main

import (
	"strconv"
)

func addDigits(num int) int {
	for num >= 10 {
		num = sumDigits(num)
	}
	return num
}

func sumDigits(x int) int {
	str, sum := strconv.Itoa(x), 0
	for _, x := range str {
		sum += int(x - '0')
	}
	return sum
}
